### PR TITLE
Handwashing action bar for motives-enabled play

### PR DESCRIPTION
**Pull request by Kubius**
[FEAT]
## About the PR 

Adds a repeating handwashing action used when a motives-enabled character washes their hands; this action requires two unoccupied hands and physical proximity (as one would expect), and applies roughly at the same rate as spam-click handwashing already would.

Absolute rate of washing is still fairly low (2 hygiene per cycle at a baseline, unchanged from before); those with medical or chef training will receive 3 hygiene per cycle. The bar is not visible to other people.

Along with this addition, handwashing's ability to raise your hygiene from very low levels has been curbed; hygiene dropping below SIMS_HYGIENE_THRESHOLD_MESSY (a new define, set to 50% at the moment) will prevent you from regaining hygiene using handwashing.

## Why's this needed? 

Handwashing being slow is understandable (since the quick recovery to hygiene is supposed to be showering), but making it an automatically repeating action cuts back on the tedium and replaces it with an opportunity to keep chatting with people since your (real world) hands are free. The hygiene limit ensures that aforementioned showering is the way to get clean if you're properly filthy.

## Changelog 

```changelog
(u)Kubius
(*)Handwashing update for RP: Washing your hands now automatically loops, and is more effective for medical or culinary personnel, but cannot raise your hygiene if it has fallen below half.
```

### CODE REVIEW

**Comment by MarkNais** [comment]
In my head I always thought handwashing should be limited to getting between 20% and 50% "clean"; I would like the opinion of others on that.

File: code/obj/submachine/cooking.dm [hunk 1]
```diff
@@ -68,13 +68,67 @@
...
+	var/obj/submachine/chef_sink/sink
+
+	New(usermob,sinkerino)
+		user = usermob
+		sink = sinkerino
+		..()
+
+	onUpdate()
+		..()
+		if(BOUNDS_DIST(user, sink) > 1 || user == null || sink == null || user.l_hand || user.r_hand)
```
Comment (MarkNais) [inline code/obj/submachine/cooking.dm:95]: For thrice-repeated conditions like this I personally tend to make a `runCondition(..)` proc. (or a better name)
`if(!runConditions(user, sink))` etc

**Review by MarkNais — APPROVED** [review]

**Comment by thegrb93** [comment]
> In my head I always thought handwashing should be limited to getting between 20% and 50% "clean"; I would like the opinion of others on that.

This was also a suggestion I added in the previous PR but the author was very argumentative about it.

**Comment by Kubius** [comment]
Perhaps a cleanliness threshold (say, 50%) below which you can no longer gain hygiene from handwashing? I can add such a check in here, or in a subsequent PR if that's preferred to keep the balance considerations separate.

**Comment by thegrb93** [comment]
Pretty good idea

**Comment by MarkNais** [comment]
Add the 50% limit in this PR please. A small sample of other admins agreed to it's inclusion.

**Comment by Kubius** [comment]
Threshold now introduced. Checked ingame; behavior both above and below the messy hygiene threshold appears to be working as intended.

**Review by MarkNais — APPROVED** [review]

**Review by pali6 — APPROVED** [review]

**Pull request merged**